### PR TITLE
feat(query): add "Schema Default Invalid" for OpenAPI

### PR DESCRIPTION
**Pull request by rafaela-soares**
Signed-off-by: Rafaela Soares rafaela.soares@checkmarx.com

**Proposed Changes**
- Added "Schema Default Invalid" query for OpenAPI. It checks if the 'default' field is not consistent with the schema's type

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/libraries/openapi/library.rego [hunk 1]
```diff
@@ -105,3 +105,22 @@ undefined_field_in_numeric_schema(value, field) {
 	is_numeric_type(value.type)
 	object.get(value, field, "undefined") == "undefined"
 }
+
+# It verifies if the 'field' is consistent with the 'type'
+invalid_field(field, type) {
+	numeric := {"integer", "number"}
+	type == numeric[_]
```
Comment (felipe-avelar) [inline assets/libraries/openapi/library.rego:112]: I think you can use `is_numeric_type` here, no?

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Besides a little comment, it looks good to me. But I think this query could have a positive test with array and object...

File: assets/queries/openAPI/schema_default_invalid/metadata.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "a96bbc06-8cde-4295-ad3c-ee343a7f658e",
+  "queryName": "Schema Default Invalid",
+  "severity": "INFO",
+  "category": "Insecure Configurations",
```
Comment (felipe-avelar) [inline assets/queries/openAPI/schema_default_invalid/metadata.json:5]:
```suggestion
  "category": "Structure and Semantics",
```

File: assets/queries/openAPI/schema_default_invalid/query.rego [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+
+	[path, value] := walk(doc)
+
+	openapi_lib.invalid_field(value["default"], value.type)
```
Comment (felipe-avelar) [inline assets/queries/openAPI/schema_default_invalid/query.rego:11]: What happens if `default` is not defined?

**Review by felipe-avelar — CHANGES_REQUESTED** [review]

File: assets/queries/openAPI/schema_default_invalid/query.rego (hunk 3 above)
Comment (rafaela-soares) [reply assets/queries/openAPI/schema_default_invalid/query.rego:11]: The query does not return any result

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**